sound/soc/sof/intel/hda-stream.c

```diff
@@ -322,8 +322,12 @@ int hda_dsp_stream_hw_params(struct snd_sof_dev *sdev,
 	/* decouple host and link DMA */
 	mask = 0x1 << hstream->index;
 	snd_sof_dsp_update_bits(sdev, HDA_DSP_PP_BAR, SOF_HDA_REG_PP_PPCTL,
+#ifndef CONFIG_SND_SOC_SOF_FORCE_LEGACY_HDA
```

Comment:
again use a variable (or a field in the device structure).

Reply:
you mean the same as above with adding dsp availability check?